### ISSUE TITLE
Warning is not issued for uninitialized non-nullable reference type fields in structs

### ISSUE THREAD

**Issue by sfuller** [dotnet/roslyn#39580]
**Version Used**: 
 Version:   3.0.100
 Commit:    04339c3a26

**Steps to Reproduce**:
Compile the following code:
```cs
#nullable enable

public struct Foo
{
    public string Value;
    
    public int DoStuff()
    {
        return Value.Length;
    }
}
```

**Expected Behavior**:
A CS8618 or similar warning is issued due to `Value` possibly not being initialized.
Example: `warning CS8618: Non-nullable field 'Value' is uninitialized. Consider declaring the field as nullable.`

**Actual Behavior**:
No warnings are issued.

**Note**
When `Foo` is changed from a struct to a class, the warning is reported.
I was able to reproduce this with sharp lab: [link](https://sharplab.io/#v2:D4AQTAjAsAULDEA7ArgG1QQwEaoKYAJdFs9ZYQBmfAZwBcAnZAY1vwDEB7D2Ab1nwH5KQiAAZ8ANQypkuANz9BigcICWiVgBEOAZVrIAZgYAUASmX4+MQTaEB2SdNkA6ADJEA5rQAWC64IBfWACgA===)

**Comment by jaredpar** [comment]
This behavior is currently "By Design". This was an explicit decision where we traded off correctness for convenience in the language. 

Moving to csharplang as it's a language issue.
